hardware/esp8266com/esp8266/libraries/ESP8266WebServer/src/ESP8266WebServer.cpp

```diff
@@ -41,6 +41,10 @@ ESP8266WebServer::ESP8266WebServer(int port)
 }
 
 ESP8266WebServer::~ESP8266WebServer() {
+  if (_headerKeys) {
+    free(_headerKeys);
+    _headerKeys = 0;
```

Comment:
I doubt it because the object is being destroyed anyway but why not, can't be too safe. Does the rest of the PR work for you?